crates/symbolicator-service/src/services/symbolication/sourcemap.rs

```diff
@@ -0,0 +1,451 @@
+                                frame_source_mapping.sourcemap_artifact =
+                                    Some(sourcemap_artifact.to_owned());
+                                continue;
+                            }
+                        }
+                    }
+                }
+            }
+
+            if let Some(source_file) = frame_source_mapping.source_file.as_mut() {
+                if let Some(sourcemap_file) = frame_source_mapping.sourcemap_file.as_mut() {
+                    frame_source_mapping.smcache =
+                        Some(smcache_from_files(source_file, sourcemap_file));
+                }
+            }
+
+            collected_artifacts.insert(abs_path, frame_source_mapping);
+        }
+
+        collected_artifacts
+    }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn js_processing_symbolicate(
+        &self,
+        request: JsProcessingSymbolicateStacktraces,
+    ) -> Result<JsProcessingCompletedSymbolicationResponse, anyhow::Error> {
+        let artifacts = self.collect_stacktrace_artifacts(&request).await;
+
+        let stacktraces: Vec<_> = request
+            .stacktraces
+            .into_iter()
+            .map(|trace| js_processing_symbolicate_stacktrace(trace, &artifacts))
+            .collect();
+
+        Ok(JsProcessingCompletedSymbolicationResponse {
+            stacktraces,
+            ..Default::default()
+        })
+    }
+}
+
+fn js_processing_symbolicate_stacktrace(
+    stacktrace: JsProcessingRawStacktrace,
+    artifacts: &HashMap<String, FrameSourceMapping>,
+) -> CompleteJsStacktrace {
+    let mut symbolicated_frames = vec![];
+    let unsymbolicated_frames_iter = stacktrace.frames.into_iter();
+
+    for mut frame in unsymbolicated_frames_iter {
+        match js_processing_symbolicate_frame(&mut frame, &artifacts) {
+            // TODO: Should frames filtering (eg. unsupported platform, or non-in_app)
+            // be done here or by monolith?
+            Ok(frame) => symbolicated_frames.push(frame),
+            Err(_status) => {
+                symbolicated_frames.push(JsProcessingSymbolicatedFrame {
+                    status: FrameStatus::Missing,
+                    raw: frame,
+                });
+            }
+        }
+    }
+
+    CompleteJsStacktrace {
+        frames: symbolicated_frames,
+    }
+}
+
+fn js_processing_symbolicate_frame(
+    frame: &mut JsProcessingRawFrame,
+    artifacts: &HashMap<String, FrameSourceMapping>,
+) -> Result<JsProcessingSymbolicatedFrame, FrameStatus> {
+    if let Some(artifact) = artifacts.get(&frame.abs_path) {
+        if let Some(smcache) = &artifact.smcache {
+            let token = smcache
+                .get()
+                .lookup(SourcePosition::new(
+                    frame.lineno.unwrap() - 1,
+                    frame.colno.unwrap() - 1,
+                ))
+                .unwrap();
+
+            let file = token.file().unwrap();
```

Comment:
there is a lot of `unwrap` all over the place :-) We should decide how to do error handling here.
What does the current code do if some source was not fetched successfully? I believe it is reporting these to the user somehow?